internal/service/federatedsettingsidentityprovider/model_federated_settings_identity_provider.go

```diff
@@ -4,12 +4,73 @@ import (
 	"sort"
 	"strings"
 
-	admin20231115008 "go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.mongodb.org/atlas-sdk/v20231115014/admin"
 
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+	"github.com/spf13/cast"
 )
 
-func FlattenFederatedSettingsIdentityProvider(federatedSettingsIdentityProvider []admin20231115008.FederationIdentityProvider) []map[string]any {
+const WORKFORCE = "WORKFORCE"
+
+func ExpandIdentityProviderOIDCCreate(d *schema.ResourceData) *admin.FederationOidcIdentityProviderUpdate {
+	return &admin.FederationOidcIdentityProviderUpdate{
+		Audience:          conversion.StringPtr(d.Get("audience").(string)),
+		AssociatedDomains: expandAssociatedDomains(d),
+		AuthorizationType: conversion.StringPtr(d.Get("authorization_type").(string)),
+		ClientId:          conversion.StringPtr(d.Get("client_id").(string)),
+		Description:       conversion.StringPtr(d.Get("description").(string)),
+		DisplayName:       conversion.StringPtr(d.Get("name").(string)),
+		GroupsClaim:       conversion.StringPtr(d.Get("groups_claim").(string)),
+		IdpType:           conversion.StringPtr(WORKFORCE),
```

Comment:
This hardcoded value will be adjusted in follow up PRs when support for workload oidc is added right?

Reply:
Exactly